Plugins/Json/src/SurfaceJsonConverter.cpp

```diff
@@ -42,6 +42,11 @@ void Acts::to_json(nlohmann::json& j, const Acts::Surface& surface) {
   j = SurfaceJsonConverter::toJson(gctx, surface);
 }
 
+void Acts::to_json(nlohmann::json& j, const Acts::Surface* surface) {
+  Acts::GeometryContext gctx;
+  j = SurfaceJsonConverter::toJson(gctx, *surface);
```

Comment:
This needs to check if the surface does not have an associated detector element, otherwise, this is invalid.

Reply:
Mmm, we have the same functionality for `std::shared_ptr<Surface>` - this just mirrors it for plain pointers, shall I change that then as well?

I guess we could throw and exception? That the default context is used is already pointed out in the comments.

This is needed because the `GeometryIdHierarchyMap` converter relies on the `nlohmann::json` bare `to_json` implementation.

Reply:
Didn't we add a way to serialize the surface with an explicit context? The problem with a default constructed context is that it's up to the detector element how it's handled. 

Sort of fundamentally, surfaces with detector elements can't really be serialized by writing out the transform, since it's only a snapshot. For the podio Track edm, I explicitly use the identifier of the surface if it has one, so it can be tied back to the detector element.